git_machete/git_operations.py

```diff
@@ -167,8 +168,8 @@ def __init__(self) -> None:
         self.__initial_log_shas_cached: Dict[FullCommitHash, List[FullCommitHash]] = {}
         self.__remaining_log_shas_cached: Dict[FullCommitHash, List[FullCommitHash]] = {}
         self.__reflogs_cached: Optional[Dict[AnyBranchName, Optional[List[ReflogEntry]]]] = None
-        self.__merge_base_cached: Dict[Tuple[FullCommitHash, FullCommitHash], FullCommitHash] = {}
-        self.__contains_equivalent_tree_cached: Dict[Tuple[FullCommitHash, FullCommitHash], bool] = {}
+        self.__merge_base_cached: Dict[ForkPointOverrideData, FullCommitHash] = {}
```

Comment:
🤔

Reply:
This type is indeed equivalent... but logically this key has nothing to do with fork point override data!

Reply:
In this case, this is just a pair of commits, with no underlying meaning of one commit being `to` and the other being `whileDescendantOf`